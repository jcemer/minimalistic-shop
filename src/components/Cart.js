import React, { PropTypes } from 'react'
import _map from 'lodash/map'
import _isEmpty from 'lodash/isEmpty'

import CartItem from './CartItem'

const Cart = ({ purchased, items, checkout }) => {
  if (purchased) {
    return (
      <div className="ui message green">Well done. Thank you!</div>
    )
  }

  if (_isEmpty(items)) {
    return (
      <div className="ui message yellow">It's empty. Let's shop!</div>
    )
  }

  return (
    <div>
      <div className="ui unstackable divided items">
        {_map(items, item => <CartItem key={item._id} {...item} />)}
      </div>
      <button className="ui button fluid blue" onClick={checkout}>
        Checkout
      </button>
    </div>
  )
}

Cart.propTypes = {
  purchased: PropTypes.bool,
  items: PropTypes.objectOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      quantity: PropTypes.number.isRequired,
    }).isRequired
  ).isRequired,
  checkout: PropTypes.func.isRequired
}

export default Cart
