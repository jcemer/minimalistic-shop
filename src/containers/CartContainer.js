import React from 'react'
import { connect } from 'react-redux'

import { checkout } from '../actions/cart'
import Cart from '../components/Cart'

export const CartContainer = (props) => {
  return (
    <div>
      <div className="ui header">Shopping Cart</div>
      <Cart {...props} />
    </div>
  )
}

export const mapStateToProps = (
  { cart: { purchased, items } }
) => ({ purchased, items })

export default connect(mapStateToProps, { checkout })(CartContainer)
