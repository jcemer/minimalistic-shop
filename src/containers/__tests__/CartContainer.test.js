import React from 'react'
import { shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import { CartContainer, mapStateToProps } from '../CartContainer'
import Cart from '../../components/Cart'

describe('CartContainer', () => {
  const checkout = () => {}

  describe('<Component />', () => {
    it('should render Cart with items', () => {
      const items = { '123': { _id: '123', image: 'url', quantity: 2 } }
      const wrapper = shallow(<CartContainer checkout={checkout} items={items} />)

      expect(wrapper.contains(<Cart checkout={checkout} items={items} />)).toBe(true)
    })

    it('should render the proper HTML of Cart with items', () => {
      const items = { '123': { _id: '123', image: 'url', quantity: 2 } }
      const wrapper = render(<CartContainer checkout={checkout} items={items} />)

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should returns items from cart', () => {
      const props = mapStateToProps(
        { cart: { items: {} } },
      )

      expect(props).toEqual({ items: {} })
    })
  })
})
