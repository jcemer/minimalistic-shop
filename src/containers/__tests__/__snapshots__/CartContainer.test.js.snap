exports[`CartContainer <Component /> should render the proper HTML of Cart with items 1`] = `
<div>
  <div
    class="ui header">
    Shopping Cart
  </div>
  <div>
    <div
      class="ui unstackable divided items">
      <div
        class="ui item">
        <div
          class="ui tiny image">
          <img
            alt="Product 123"
            src="url" />
        </div>
        <div
          class="middle aligned content">
          <div
            class="ui tiny header">
            Product 123
          </div>
          <div>
            Quantity: 2
          </div>
        </div>
      </div>
    </div>
    <button
      class="ui button fluid blue">
      Checkout
    </button>
  </div>
</div>
`;
